File: Scripts/World.cs
using Godot;
using static StateFunctions;

public class World : Node
{
    private Label label;
    private State current_state;

    public override void _Ready()
    {
        current_state = InitialState(GetChildren());
    }

    public override void _Process(float delta)
    {
        current_state = UpdateState(current_state, delta, this);
    }
}

File: FSharp/World.fs
namespace FSharp

open Godot
open State
open Game

type World() =
    inherit Node2D()

    let mutable currentState = Game.state
    //private State current_state;

    override this._Ready () =
        GD.Print "Hello from F#!"
        currentState <- _ready this currentState
        printState currentState

    override this._Process delta =
        GD.Print delta
        currentState <- _process currentState